game-core/src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java

```diff
@@ -413,6 +423,130 @@ public void resetToDefault() {
+        private final String displayName;
+        @Getter
+        @Nonnull
+        private final String host;
+        @Getter
+        @Nonnull
+        private final int port;
+        @Getter
+        @Nonnull
+        private final boolean isEncrypted;
+
+        @Override
+        public String toString() {
+          return displayName;
+        }
+      }
+
+      private final List<EmailProviderSetting> knownProviders = Arrays.asList(
+          new EmailProviderSetting("GMail", "smtp.gmail.com", 587, true),
+          new EmailProviderSetting("Hotmail", "smtp.live.com", 587, true)
+      );
+
+      final JTextField serverField = new JTextField(hostSetting.getValue().orElse(""), 20);
+
+      final JSpinner portSpinner = new JSpinner(new SpinnerNumberModel(
+          (int) portSetting.getValue().orElse(0), 0, 65535, 1));
+
+      final JCheckBox tlsCheckBox = new JCheckBox("SSL/TLS", tlsSetting.getValue().orElse(true));
+
+      final JTextField usernameField = new JTextField(usernameSetting.getValue().orElse(""), 20);
+
+      final JPasswordField passwordField = new JPasswordField(passwordSetting.getValue().orElse(""), 20);
+
+      final JPanel panel = JPanelBuilder.builder()
+          .verticalBoxLayout()
+          .addLeftJustified(new JLabel("Email Server"))
+          .addLeftJustified(serverField)
+          .addLeftJustified(new JLabel("Port"))
+          .addLeftJustified(JPanelBuilder.builder()
+              .horizontalBoxLayout()
+              .addLeftJustified(portSpinner)
+              .addLeftJustified(tlsCheckBox)
+          .build())
+          .addVerticalStrut(5)
+          .addLeftJustified(JButtonBuilder.builder()
+              .title("Presets...")
+              .actionListener(() -> {
+                final JComboBox<EmailProviderSetting> comboBox =
+                    JComboBoxBuilder.builder(EmailProviderSetting.class)
+                        .items(knownProviders)
+                        .build();
+                if (JOptionPane.showConfirmDialog(this.panel.getParent(), JPanelBuilder.builder().add(comboBox).build(),
+                    "Select a Preset", JOptionPane.OK_CANCEL_OPTION) == JOptionPane.OK_OPTION) {
+                  final EmailProviderSetting config = (EmailProviderSetting) comboBox.getSelectedItem();
+                  serverField.setText(config.getHost());
+                  portSpinner.setValue(config.getPort());
+                  tlsCheckBox.setSelected(config.isEncrypted());
+                }
+              })
+            .build())
+          .addLeftJustified(new JLabel("Username"))
+          .addLeftJustified(usernameField)
+          .addLeftJustified(new JLabel("Password"))
+          .addLeftJustified(passwordField)
+          .build();
+
+      @Override
+      public JComponent getUiComponent() {
+        return panel;
+      }
+
+      @Override
+      public Map<GameSetting<?>, Object> readValues() {
+        final char[] password = passwordField.getPassword();
```

Comment:
Just thinking ahead to when you have to do something similar in the JFX UI...  In the future, it might be nice to introduce a general helper someplace like `withSensitiveCharArray(Supplier<char[]>, Function<char[], R>)` that would basically implement the _Execute-Around_ pattern:

```java
<R> R withSensitiveCharArray(final Supplier<char[]> supplier, final Function<char[], R> function) {
  final char[] value = supplier.get();
  try {
    return function.apply(value);
  } finally {
    Arrays.fill(value, '\0');
  }
}
```

And then call it like:

```java
return withSensitiveCharArray(passwordField::getPassword, password -> {
    // build map
    return map;
});
```